common/rpc/rpc.go

```diff
@@ -235,3 +253,103 @@ func (d *RPCFactory) dial(hostName string, tlsClientConfig *tls.Config) *grpc.Cl
 func (d *RPCFactory) GetTLSConfigProvider() encryption.TLSConfigProvider {
 	return d.tlsFactory
 }
+
+// CreateLocalFrontendHTTPClient gets or creates a cached frontend client.
+func (d *RPCFactory) CreateLocalFrontendHTTPClient() (*common.FrontendHTTPClient, error) {
+	return d.localFrontendClient()
+}
+
+// createLocalFrontendHTTPClient creates an HTTP client for communicating with the frontend.
+// It uses either the provided frontendURL or membership to resolve the frontend address.
+func (d *RPCFactory) createLocalFrontendHTTPClient() (*common.FrontendHTTPClient, error) {
+	// dialer and transport field values copied from http.DefaultTransport.
+	dialer := &net.Dialer{
+		Timeout:   30 * time.Second,
+		KeepAlive: 30 * time.Second,
+	}
+	transport := &http.Transport{
+		Proxy:                 http.ProxyFromEnvironment,
+		DialContext:           dialer.DialContext,
+		ForceAttemptHTTP2:     true,
+		MaxIdleConns:          100,
+		IdleConnTimeout:       90 * time.Second,
+		TLSHandshakeTimeout:   10 * time.Second,
+		ExpectContinueTimeout: 1 * time.Second,
+	}
+	client := http.Client{}
+
+	// Default to http unless TLS is configured.
+	scheme := "http"
+	if d.frontendTLSConfig != nil {
+		transport.TLSClientConfig = d.frontendTLSConfig
+		scheme = "https"
+	}
+
+	var address string
+	service, err := extractServiceFromURL(d.frontendHTTPURL)
+	if err != nil {
+		return nil, err
```

Comment:
nit: Could you add a simple log message for this and other possible errors during client creation.

Reply:
Hmmm... I think this is fine since fx will log this error when server startup fails.